### ISSUE TITLE
Refresh Predefined Setups - Programmable Post-Processing

### ISSUE THREAD

**Issue by riverwanderer** [vassalengine/vassal#12260]
A new component for module maintenance only. Like the Startup GKC but only executed when a user (module dev) checks a new box in the Refresh Predefined Setups tool e.g."Custom Post-Processing".

My hope is that, immediately prior to saving each PDS, one or more Refresh GKCs will execute before the game is saved.

Implementation might be as another Startup GKC drop-down option "Only after Refresh of Predefined Setup" but no other Startup GKCs should be executed during or after a Refresh.

Use case 1: Modules with lots of Predefine Setup files. Avoid adding to game startup time where each file needs changes that can only be done manually or currently via a Startup GKC.

Use case 2: Establishing a new deck or At-Start Stack. Similar to case 1 but with the additional benefit that the module dev will be able to reclaim module/log file size once the special refresh has been successfully completed - because such a function will often require a trait on many pieces, which could now be removed. Temporary Piece Palettes established for this purpose can also be removed, keeping the module clean(er).

**Comment by riverwanderer** [comment]
Reverted after accidental merge. New PR submitted.
